docs/source/torch_cuda_memory.rst

```diff
@@ -54,6 +63,30 @@ The address b7f064c000000_0 refers to the (b)lock at address 7f064c000000 which
 This unique string can be looked up in the Active Memory Timeline and searched
 in the Active State History to examine the memory state when a tensor was allocated or freed.
 
+.. _non_pytorch_alloc:
+
+Identifying Non-PyTorch allocations
```

Comment:
we already have a util that does this, `torch.cuda.device_memory_used` that also correctly handles device reordering that might happen with CUDA_VISIBLE_DEVICES, which won't work with naive application of the snippet below

Reply:
TIL! ok, i will update to reference that instead.  

IIUC your point is that if i pass '0' to pynvml, it will spawn a new process that escapes CUDA_VISIBLE_DEVICES and it may see a different device0 as the current torch process sees?

Reply:
btw is this doc correct?  is .memory_usage() giving us stats about memory bw utilization? or is it actually reporting the current allocation?

<img width="1106" alt="image" src="https://github.com/user-attachments/assets/79a33f36-5a83-44c2-8165-4b708a3e7ae1" />

Reply:
`CUDA_DEVICE_ORDER` can impact this as well. please see:

https://github.com/stas00/ipyexperiments/blob/569a450e204b1da60e5ef07a96c91553b286ea14/docs/ipyexperiments.md#different-ordering-of-gpu-ids

Reply:
Yeah we should be doing this based on uuid, and not on parsing env vars :-(

Reply:
pynvml doesn't spawn new processes, but when doing `pynvml.nvmlDeviceGetHandleByIndex(device_idx)` it will ignore whatever reordering runtime was seeing due to CUDA_DEVICE_ORDER and CUDA_VISIBLE_DEVICES and just go with pci order. In any case, using pytorch utilities is better than raw pynvml code that you have here.

Reply:
updated it to show the torch API.